core/kazoo_number_manager/src/knm_numbers.erl

```diff
@@ -487,10 +486,7 @@ new(Options, ToDos, KOs) -> new(Options, ToDos, KOs, 'not_reconcilable').
 new(Options, ToDos, KOs, Reason) ->
     #{'todo' => ToDos
      ,'ok' => []
-     ,'ko' => case is_function(Reason, 1) of %%FIXME: find something better than Reason/1.
-                  'false' -> maps:from_list([{KO, Reason} || KO <- KOs]);
-                  'true' -> maps:from_list([{KO, Reason(KO)} || KO <- KOs])
-              end
+     ,'ko' => maps:from_list([{KO, Reason} || KO <- KOs])
```

Comment:
Same here, are you fixing a bug?

Reply:
I faced this same issue when I started using _dialyze-hard_ instead of _dialyze-changed_ as it was not catching most of the warning, even the most evident ones.

As part of fixing those warning, I searched around _core/kazoo_number_manager_ application and I could not find any _Reason_ that would cause that case statement to go through the "true" branch.

I am sorry for resurrecting this very old comment, I just saw it and wanted to add my findings 😄